### PR TITLE
WT-13204 Add diagram for block layout in block manager architecture guide

### PR DESCRIPTION
**Pull request by Sean04**

### CODE REVIEW

**Review by ruby-oujo — APPROVED** [review]

File: src/docs/arch-block.dox [hunk 1]
```diff
@@ -15,6 +15,35 @@ re-written, they are written to new locations in the file. The size of a block
...
+@startuml{block.png}
+
+title Block Layout
+node {
+ json "Block" as J {
+    "Page Header [00-27]": {
+      "recno (column store)": "00-07",
+      "write_gen": "08-15",
+      "mem_size": "16-19",
+      "entries": "20-23",
```
Comment (etienneptl) [inline src/docs/arch-block.dox:28]: We mention those two in the code:
```
        uint32_t entries; /* 20-23: number of cells on page */
        uint32_t datalen; /* 20-23: overflow data length */
```

**Review by etienneptl — APPROVED** [review]
LGTM! One comment

File: src/docs/arch-block.dox (hunk 1 above)
Comment (Sean04) [reply src/docs/arch-block.dox:28]: good catch

Comment (Sean04) [reply src/docs/arch-block.dox:28]: Not sure the best way to format this. What do you think of 
```
      "entries (standard pages)": "20-23",
      "datalen (overflow pages)": "20-23",
```
  or
```
	"entries (or datalen for overflow pages)": "20-23",
```

Comment (etienneptl) [reply src/docs/arch-block.dox:28]: I like the second one. The first one may make you think it's two different things and we put the wrong numbers.

Comment (Sean04) [reply src/docs/arch-block.dox]: Fixed 6e000f7de1

File: src/docs/arch-block.dox [hunk 2]
```diff
@@ -15,6 +15,35 @@ re-written, they are written to new locations in the file. The size of a block
...
+      "recno (column store)": "00-07",
+      "write_gen": "08-15",
+      "mem_size": "16-19",
+      "entries (or datalen for overflow pages)": "20-23",
+      "type": "24",
+      "flags": "25",
+      "unused (padding)": "26",
+      "version": "27"
+    },
+    "Block Header [28-40]": {
```
Comment (tod-johnson-mongodb) [inline src/docs/arch-block.dox:34]: Relate this to the code by mentioning types here. I think the Block Header is WT_BLOCK_HEADER.

**Review by tod-johnson-mongodb — APPROVED** [review]
LGTM - I have one suggestion about relating this to the code.

File: src/docs/arch-block.dox [hunk 3]
```diff
@@ -15,6 +15,35 @@ re-written, they are written to new locations in the file. The size of a block
 is a multiple of the allocation size which is set during creation of the
 associated WiredTiger data file see: WT_SESSION::create.
 
+@plantuml_start{block.png }
+@startuml{block.png}
+
+title Block Layout
```
Comment (tod-johnson-mongodb) [inline src/docs/arch-block.dox:21]: If you write this as: Diagram: Block Layout it becomes easy to search for all diagrams in the manual.

Comment (Sean04) [reply src/docs/arch-block.dox:21]: Are you referring to the search bar in the docs webpage or just a way to search the docs source code?

This corresponds to the text at the top of the generated diagram. Is this what you're suggesting?

Screenshot 2024-07-01 at 10 26 52 AM

Comment (tod-johnson-mongodb) [reply src/docs/arch-block.dox:21]: * Both. If we have a standard way of titling diagrams it makes searching for all of them easy wherever that search is done.
* Yes, that is what I meant. Then searching for Diagram: will find all diagrams.

**Pull request merged**